src/Prover-Axelar.sol

```diff
@@ -51,6 +52,15 @@ contract DealClient is AxelarExecutable {
     mapping(bytes => Status) public pieceStatus;
     mapping(bytes => uint256) public providerGasFunds; // Funds set aside for calling oracle by provider
     mapping(uint256 => DestinationChain) public chainIdToDestinationChain;
+    event DealNotify(
```

Comment:
Overall I think this change should be in a different contract.  "PoRepService" or "BuiltinMarketExtension" or with some better name.  The reason is that the original contract is orthogonally useful for use cases of cross chain storage that want to handle any extensions to the deal protocol in some cross chain onramp contract.  To be concrete if I want to pay ERC20s for filecoin deal I can 
1. Do that in the onramp contract cross chain by interacting with a smart contract 
2. Do that offchain by signing a message that gets validated as you are doing here

One exception is the receiveDataCap handling here which is more of a bug fix for Prover-Axelar contract.